server/routes/flight.js

```diff
@@ -1,92 +1,120 @@
+	const outDate = validator.isDate(outboundDate, validatorOptions)
+		? outboundDate
+		: "";
+	const inDate = validator.isDate(inboundDate, validatorOptions)
+		? inboundDate
+		: "";
+	// test for errors
+	const errors = {};
+	const origin = from.includes("-sky")
+		? from
+		: (errors.from = "Invalid format or missing value.");
+	const destination = to.includes("-sky")
+		? to
+		: (errors.to = "Invalid format or missing value.");
+	const departDate = outDate
+		? !dateInPast(outDate)
+			? outboundDate
+			: (errors.outboundDate = "Departure date is a past date.")
+		: (errors.outboundDate = "Invalid date format.");
 
-//Get All available currencies
-//Example: /api/flights/currencies
-router.get("/currencies", (req, res, next) => {
-  unirest
-    .get(
-      "https://skyscanner-skyscanner-flight-search-v1.p.rapidapi.com/apiservices/reference/v1.0/currencies"
-    )
-    .headers({
-      "x-rapidapi-key": rapidApiKey,
-      "x-rapidapi-host": rapidApiHost,
-      useQueryString: true,
-    })
-    .then((response) => {
-      res.send(response.body);
-    })
-    .catch((err) => {
-      next(err);
-    });
+	if (Object.keys(errors).length === 0) {
+		const request = unirest.get(
+			`https://skyscanner-skyscanner-flight-search-v1.p.rapidapi.com/apiservices/browsequotes/v1.0/US/USD/en-US/${origin}/${destination}/${departDate}`
+		);
+		request.headers({
+			"x-rapidapi-key": rapidApiKey,
+			"x-rapidapi-host": rapidApiHost,
+			useQueryString: true,
+		});
+		// API does not return results for inboundDate
+		if (inboundDate) {
+			const returnDate = inDate
+				? !dateInPast(inDate, outDate)
+					? inboundDate
+					: ""
+				: "";
+
+			request.query({
+				query: returnDate,
+			});
+		}
+		request
+			.then((response) => {
+				res.json(response.body);
+			})
+			.catch((err) => {
+				next(err);
+			});
+	} else {
+		res.send(errors);
+	}
 });
 
-//Get All available places
-//Example : /api/flights/places/france/fr/eur/fr.eu/
-router.get(
-  "/places/:country/:countryCode/:currency/:locale",
-  (req, res, next) => {
-    const { country, countryCode, currency, locale } = req.params;
-    unirest
-      .get(
-        `https://skyscanner-skyscanner-flight-search-v1.p.rapidapi.com/apiservices/autosuggest/v1.0/${countryCode}/${currency}/${locale}/`
-      )
-      .headers({
-        "x-rapidapi-key": rapidApiKey,
-        "x-rapidapi-host": rapidApiHost,
-        useQueryString: true,
-      })
-      .query({
-        query: country,
-      })
-      .then((response) => {
-        res.send(response.body);
-      })
-      .catch((err) => {
-        next(err);
-      });
-  }
-);
+router.get("/places/:regionId", (req, res, next) => {
+	const { regionId } = req.params;
+	// regionId can be either the full name or abbreviation of the state, same results
+	const error = {};
+	const region = validator.isAlpha(regionId)
+		? regionId
+		: (error.region = "Region not in proper format.");
+
+	if (Object.keys(error).length === 0) {
```

Comment:
This is another error handling situation that you could probably abstract with the others above into an "error handling function" of some kind.

A small personal preference/tip: whenever I find myself having to check if an object has any properties in it, I prefer to use a [`Map`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map). It works in a similar way to an object, except it is optimized for adding and removing properties, and it has some useful methods and properties built-in, like `.size`, which would allow you to do something like this:

```javascript
if (errorMap.size === 0) { 
  // ... 
}
```